python/tk_nuke/__init__.py

```diff
@@ -70,123 +74,153 @@ def __create_tank_error_menu():
             __create_tank_disabled_menu(e)
             return
-        
+
         # try to get current ctx and inherit its values if possible
         curr_ctx = None
         if tank.platform.current_engine():
             curr_ctx = tank.platform.current_engine().context
-        
+
         # and now extract a new context based on the file
         new_ctx = tk.context_from_path(file_name, curr_ctx)
-        
+        logger.debug("New context computed to be: %r" % new_ctx)
+
         # now restart the engine with the new context
         __engine_refresh(tk, new_ctx)
     except Exception, e:
+        logger.exception("An exception was raised during addOnScriptSave callback.")
         __create_tank_error_menu()
 
 
-def tank_startup_node_callback():    
+def tank_startup_node_callback():
     """
-    Callback that fires every time a node gets created.
-    
+    Callback that fires every time a script is loaded.
+
     Carefully manage exceptions here so that a bug in Tank never
-    interrupts the normal workflows in Nuke.    
-    """    
-    try:    
+    interrupts the normal workflows in Nuke.
+    """
+    try:
+        logger.debug("SGTK Callback: addOnScriptLoad")
         if nuke.root().name() == "Root":
+
             # file->new
             # base it on the context we 'inherited' from the prev session
             # get the context from the previous session - this is helpful if user does file->new
+            logger.debug("File->New detected.")
             project_root = os.environ.get("TANK_NUKE_ENGINE_INIT_PROJECT_ROOT")
+            logger.debug("Will spin up a new tk session based on "
+                         "previous session's config from '%s'" % project_root)
             tk = tank.Tank(project_root)
-            
+
             ctx_str = os.environ.get("TANK_NUKE_ENGINE_INIT_CONTEXT")
             if ctx_str:
                 try:
                     new_ctx = tank.context.deserialize(ctx_str)
+                    logger.debug("Will use previous context '%r'" % new_ctx)
                 except:
+                    logger.debug("Could not extract previous context "
+                                 "from env var TANK_NUKE_ENGINE_INIT_CONTEXT. "
+                                 "New session will use empty context.")
                     new_ctx = tk.context_empty()
             else:
                 new_ctx = tk.context_empty()
-    
+                logger.debug("No previous context found in TANK_NUKE_ENGINE_INIT_CONTEXT. "
+                             "New session will use empty context.")
+
         else:
             # file->open
             file_name = nuke.root().name()
-            
+            logger.debug("File->Open detected with file name '%s'" % file_name)
+
             try:
                 tk = tank.tank_from_path(file_name)
+                logger.debug("Tk instance '%r' associated with path '%s'" % (tk, file_name))
```

Comment:
Jeff picked me up on this so I'm going to raise it here :D. Small thing but it can save some performance here when debug is not enabled by switching this line to the following:
`logger.debug("Tk instance '%r' associated with path '%s'" tk, file_name)`
So that the string only gets built when if it gets logged.